client/my-sites/promote-post-i2/main.tsx

```diff
@@ -228,6 +232,33 @@ export default function PromotedPosts( { tab }: Props ) {
 		);
 	};
 
+	const renderWarningNotices = ( warnings?: string[] ) => {
+		const content = [];
+
+		for ( const item of warnings ?? [] ) {
+			switch ( item ) {
+				case 'sync_in_progress':
```

Comment:
Wonder if we should store this as a const somewhere?

Reply:
I could add some typescript support to it. Let me create a type for the warning list 👍

Reply:
Done in this commit: https://github.com/Automattic/wp-calypso/pull/94862/commits/e5737a544077a988fba886d89393de71fe6d0291

Now, the IDE will suggest the values, and we will get a warning if we try to use it without the correct type/value.